### DISCUSSION TITLE
Is gamepad/remote support possible or within scope.

### DISCUSSION THREAD

**Discussion by Quackdoc** [hecrj/iced#1896] [Q&A]
There are a few motivations for controlling software with a gamepad or TV remote style input, Currently I am primairly concerned with supporting devices like steamdeck/set-top-boxes.

**Comment by bbb651** [comment]
I'm also interested in this, I think the best option for now [`gilrs`](https://gitlab.com/gilrs-project/gilrs) and eventually switch to winit when they add gamepad support, I'm looking at making an `gamepad::Event` based on it's [`EventType`](https://docs.rs/gilrs/latest/gilrs/ev/enum.EventType.html):
- I don't think we should have filters, this seems needlessly complex and can be handle in subscriptions instead
  - and consequently drop `Dropped`, which seems like a hack anyway
- I don't like `ButtonRepeated`, I think it should instead be a `repeat` field on `ButtonPressed`
- I think force feedback is out of scope at least for now
- Buttons being `[0.0, 1.0]` ranges instead of `bool`s and triggers being buttons feels very weird to me, is there any benefit to this approach? Is it just to have joysticks be `[-1.0, 1.0]` and triggers `[0.0, 1.0]`? If so it seems very much not worth it
